src/openrct2/ride/Vehicle.cpp

```diff
@@ -1174,135 +1174,90 @@ static rct_vehicle_sound* vehicle_sounds_update_get_vehicle_sound(rct_vehicle_so
     return nullptr;
 }
 
-// Track Noises
-static void vehicle_sounds_update_sound_1(
-    Vehicle* vehicle, rct_vehicle_sound_params* sound_params, rct_vehicle_sound* sound, uint8_t panVol)
+enum class SoundType
 {
-    int32_t volume = vehicle->sound1_volume;
-    volume *= panVol;
-    volume = volume / 8;
-    volume = std::max(volume - 0x1FFF, -10000);
-
-    if (vehicle->sound1_id == SoundId::Null)
-    {
-        if (sound->sound1_id != SoundId::Null)
-        {
-            sound->sound1_id = SoundId::Null;
-            Mixer_Stop_Channel(sound->sound1_channel);
-        }
-        return;
-    }
-
-    if (sound->sound1_id != SoundId::Null && vehicle->sound1_id != sound->sound1_id)
-    {
-        Mixer_Stop_Channel(sound->sound1_channel);
-    }
+    TrackNoises,
+    OtherNoises, // e.g. Screams
+};
 
-    if ((sound->sound1_id == SoundId::Null) || (vehicle->sound1_id != sound->sound1_id))
+template<SoundType type> static uint16_t SoundFrequency(const SoundId id, uint16_t baseFrequency)
+{
+    if constexpr (type == SoundType::TrackNoises)
     {
-        sound->sound1_id = vehicle->sound1_id;
-        sound->sound1_pan = sound_params->pan_x;
-        sound->sound1_volume = volume;
-        sound->sound1_freq = sound_params->frequency;
-        uint16_t frequency = sound_params->frequency;
-        if (_soundParams[static_cast<uint8_t>(vehicle->sound1_id)][1] & 2)
+        if (_soundParams[static_cast<uint8_t>(id)][1] & 2)
```

Comment:
In a future refactor _soundParams should be modified into a set of flags. There are only 3 flags: Looping, FrequencyMod1, FrequencyMod2. Have a think about a better name than FrequencyMod...

Reply:
I prefer we have 3 functions, isLooping, trackSoundIsFrequencyModified, otherSoundIsFrequencyModified. All these functions will have switch statements, returning true or false, based on the sound id.

There are many advantages to this: the switch statements will fail in compilation if we add a new sound and don't specify them, we will avoid all those static_casts, the functions is well isolated and can be constexpr

Reply:
I'd prefer it to be an array as it's really meta data of the sound. If you see the rtd refactors we have been doing for the ride types we have been undoing functions like this and converting them into arrays of flags.

Reply:
Ok!